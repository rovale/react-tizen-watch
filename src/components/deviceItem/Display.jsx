import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ device }) => {
  const getLine = (key, a1, a2) => (
    <div key={key} className="ui-li-sub-text">
      {a1.acronym }: {a1.value}{a1.unit}
      {!!a2 && <span>,&nbsp;{a2.acronym }: {a2.value}{a2.unit}</span>}
    </div>
  );

  const getLines = () => {
    const lines = [];
    for (let i = 0; i < device.attributes.length; i += 2) {
      lines.push(getLine(i, device.attributes[i], device.attributes[i + 1]));
    }
    return lines;
  };

  return (
    <div>
      <div>{device.title}</div>
      {getLines()}
    </div>
  );
};

Display.propTypes = {
  device: PropTypes.shape({
    title: PropTypes.string.isRequired,
    attributes: PropTypes.array.isRequired,
  }).isRequired,
};

export default Display;
