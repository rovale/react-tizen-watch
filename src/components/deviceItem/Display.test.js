import React from 'react';
import { mount } from 'enzyme';

import Display from './Display';

test('it should render as expected', () => {
  const device = {
    id: 'someDevice',
    title: 'Some Device',
    attributes: [
      { acronym: 'acronym1', value: 'value 1', unit: 'unit 1' },
      { acronym: 'acronym2', value: 'value 2', unit: 'unit 2' },
      { acronym: 'acronym3', value: 'value 3', unit: 'unit 3' },
      { acronym: 'acronym4', value: 'value 4', unit: 'unit 4' },
      { acronym: 'acronym5', value: 'value 5', unit: 'unit 5' },
    ],
  };
  const wrapper = mount((<Display device={device} />)).children().first();
  expect(wrapper).toMatchSnapshot();
});
