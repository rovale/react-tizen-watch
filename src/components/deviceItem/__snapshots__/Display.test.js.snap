// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render as expected 1`] = `
<div>
  <div>
    Some Device
  </div>
  <div
    className="ui-li-sub-text"
    key="0"
  >
    acronym1
    : 
    value 1
    unit 1
    <span>
      , 
      acronym2
      : 
      value 2
      unit 2
    </span>
  </div>
  <div
    className="ui-li-sub-text"
    key="2"
  >
    acronym3
    : 
    value 3
    unit 3
    <span>
      , 
      acronym4
      : 
      value 4
      unit 4
    </span>
  </div>
  <div
    className="ui-li-sub-text"
    key="4"
  >
    acronym5
    : 
    value 5
    unit 5
  </div>
</div>
`;
